glean-core/src/upload/mod.rs

```diff
@@ -195,8 +283,21 @@ impl PingUploadManager {
             .queue
             .write()
             .expect("Can't write to pending pings queue.");
-        match queue.pop_front() {
+        match queue.front() {
             Some(request) => {
+                if let Some(rate_limiter) = &self.rate_limiter {
+                    let mut rate_limiter = rate_limiter
+                        .write()
+                        .expect("Can't write to the rate limiter.");
+                    if rate_limiter.get_state() == RateLimiterState::Throttled {
+                        log::info!(
+                            "Tried getting an upload task, but we are throttled at the moment.
```

Comment:
This will log with a newline and a lot of preceding whitespace. The second line should probably be inline here (or just be removed?)